src/driver/include/driver/WorkloadParser.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_0369D27D_9A68_4981_B344_4BAB3EE09A80_INCLUDED
+#define HEADER_0369D27D_9A68_4981_B344_4BAB3EE09A80_INCLUDED
+
+#include <boost/filesystem.hpp>
+#include <yaml-cpp/yaml.h>
+
+#include <driver/DefaultDriver.hpp>
+
+namespace genny::driver {
+
+using YamlParameters = std::map<std::string, YAML::Node>;
+namespace fs = boost::filesystem;
+
+class WorkloadParser {
```

Comment:
🤓 The first commit in this PR is to move the YAML parsing code over to this file.

Reply:
👍 This is a good separation-of-concerns. If/when #225 lands this may make sense in a separate `config` CMake module (project?). Fine to leave in driver for now.